tp030/README.md

```diff
@@ -0,0 +1,88 @@
+# TP 30: Delegates Interface
+
+## Problem Statement
+The private keys to a master DID are precious and should be kept safe. However, applications built on DWNs require messages to be signed frequently and on many devices. If the master DID private keys exist only on one device, it is onerous for remote DWNs to ping the master key holder to sign every single DWN message.
+
+### Requirements
+1. The master DID must be able to delegate a subordinate DID to sign messages on its behalf.
+2. Other DWNs must be able to verify this delegation and check that it has not been revoked.
+
+## Methods
+1. Grant
+2. Revoke
+3. Check
+  - Alternate names: StatusCheck, Status, Verify, Validate, Confirm, Audit
+4. Register
+  - Alternate names: Hook, Subscription
+
+## Characters
+For the sake of easier naming, let's define the following three DIDs who will appear throughout this TP.
+1. *Alice* is a DWN owner who will be the *delegator*.
+2. *Alice-ChatApp* is the app on Alice's phone that will be the *delegatee*.
+3. *Bob* is the *counterparty* who wants to receive DWN messages signed by Alice-ChatApp with the understanding that Alice-ChatApp is representing Alice. Bob wants to be confident that Alice has not revoked Alice-ChatApp's grant, so he will keep his eyes peeled.
+
+## Granting and Invocation
+Alice creates a `DelegatesGrant` message with her own DID in the `delegatedBy` field and Alice-ChatApp's DID in the `delegatedTo` field.
+When Alice-ChatApp sends a message to Bob invoking the grant, the `authorization` must contain both Alice-ChatApp's signature AND a reference to the `DelegatesGrant` message. The reference may take one of two forms: Either the entire `DelegatesGrant` message may be listed in the `delegatesGrant` field of the `authorization` OR the CID of the `DelegatesGrant` message may appear in the `delegatesGrantId` field of the `authorization`. The fields `delegatesGrant` and `delegatesGrantId` are mutually exclusive.
+
+## Obtaining and Processing Grants
+When Bob receives the message from Alice-ChatApp, either the message contains the entire grant in the `delegatesGrant` field OR it contains a CID of the grant `DelegatesGrant` with. In the former case, if Bob has never seen the grant before, he will process and store it in his DWN. In the latter case, he must already have the grant in his DWN.
```

Comment:
I wonder if the optionality of supporting both embedded delegate grants vs just the CID is necessary and/or useful? Developers might not want to be in the business of knowing or managing when to embed grants and when to include CIDs?

I am inclined to say we might want to support just one of the two path whichever one supports all scenarios and/or is easier to use.